src/coordinator/api/v1/handler/prometheus/native/read.go

```diff
@@ -114,89 +104,94 @@ func (h *PromReadHandler) read(reqCtx context.Context, w http.ResponseWriter, pa
 
 	// Results is closed by execute
 	results := make(chan executor.Query)
-	// Block series slices are sorted by start time
-	seriesMap := make(map[string][]block.Series)
 	go h.engine.ExecuteExpr(ctx, parser, opts, params, results)
 
+	// Block slices are sorted by start time
+	// TODO: Pooling
+	sortedBlockList := make([]block.Block, 0, 10)
 	for result := range results {
 		if result.Err != nil {
-			return nil, err
+			return nil, result.Err
```

Comment:
Would it be worth doing the error checking loop separately, before processing the blocks? Could save time if the last block in your results is erroring

Reply:
Could potentially even do the block result error check preemptively too?